src/base-api-client.ts

```diff
@@ -0,0 +1,90 @@
+import { ApiRequestHandler } from './api-request-handler';
+
+type ServiceApi = 'topic_extraction' | 'sentiment_analysis';
+
+/**
+ * Base client implementation. Intended to be extended by a specific client per API
+ */
+export class BaseApiClient<TJob, TResult> {
```

Comment:
If it's intended to be extended, should we add an abstract keyword?
```suggestion
 * Base client implementation. Intended to be extended by a specific client per API
 */
export abstract class BaseApiClient<TJob, TResult> {
```